### DISCUSSION TITLE
Why the drop-down menus in the horizontal menu stay open

### DISCUSSION THREAD

**Discussion by gokhanyildiz** [saadeghi/daisyui#1937] [Q&A] [answered]
I updated from v1 to version 3.0.3. I updated the navbar and menu component I use according to the new version. I updated from v1 to version 3.0.3. I updated the navbar and menu component I use according to the new version. But I noticed that when I click on one of the links in the submenu or an area outside the menu, the dropdown menu does not close. I have to click on the parent menu for it to close. Is there anything I need to do to make this work correctly or should I use the dropdown component instead?

Thank you in advance for your support.

**Comment by saadeghi** [comment, marked as answer]
If you use [the `<details>` tag](https://daisyui.com/components/dropdown/#dropdown-menu-using-details-tag) it works by clicking the button. click to open. click again to close.

You can also control it using JS by adding/removing the `open` attribute. That's how the `<details>` HTML element works.

If you want to use the `<details>` tag and close it when clicked outside, you need to remove the `open` attribute using JS. It's not possible for CSS to do that.

If you use [the `<label>` element](https://daisyui.com/components/dropdown/#method-2-using-label-and-css-focus) it works with CSS focus. It opens when you focus it and it closes when you focus something else. That's how CSS focus works. You can also focus the element using JS or remove the focus using JS if you want.

Let me know if you have any questions.
